Remove non-dynamic constraint on plugin versions.

staale opened the pull request:
Fixes #14704

### Context
This is currently blocking us from adopting the new plugins notation. We have a large set of projects that utilizes a common toolchain of Gradle plugins. By using a version of '1.0.+', we are able to update all projects and their build tools in a single step by just pushing a new patch version. Each dependent project then gets these fixes on their next build.

We are currently using the buildscript configuration with classpath dependencies that do have version ranges, which does allow us to set version ranges, and solves this for us. However, we would like to update to the more modern plugins notation.

I also believe this will benefit others, and does not adversely affect the current state of the code. Its just a check that is removed, without any logic changes. So this constraint doesn't really provide any value.

### Contributor Checklist
- [X] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [X] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [X] Make sure all contributed code can be distributed under the terms of the [Apache License 2.0](https://github.com/gradle/gradle/blob/master/LICENSE), e.g. the code was written by yourself or the original code is licensed under [a license compatible to Apache License 2.0](https://apache.org/legal/resolved.html).
- [X] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [x] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [X] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass sanity check: `./gradlew sanityCheck`
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:quickTest`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

jjohannes commented:
@bot-gradle test QFL

bot-gradle commented:
<!-- {"replyTargetCommentId":774033543,"teamCityBuildId":"41139300","teamCityBuildHeadRef":"951b301cf77fb252493a2e4556db8821a43dccc5","teamCityBuildStatus":"PENDING","replyApproveReviewId":null} -->
OK, I've already triggered [QuickFeedbackLinux build](https://builds.gradle.org/viewQueued.html?itemId=41139300) for you.

jjohannes commented:
Thanks @staale for pushing this. It's indeed a simple fix so we should move forward asap. I tentatively added the 7.0 milestone for consideration.

I pushed it through our TeamCity CI (ignore Travis, seems to be broken at the moment).

There are two test cases failing, which need to be adjusted (they pass now, which is ok): https://ge.gradle.org/s/iiblm52lkkpk4/tests/failed 

Would you mind adjusting them?

staale commented:
> There are two test cases failing, which need to be adjusted (they pass now, which is ok): https://ge.gradle.org/s/iiblm52lkkpk4/tests/failed
> 
> Would you mind adjusting them?

Fixed both tests, and added 2 more for the positive case, when we try to use a version range, and verifying that this works.

Is this possible for a 6.9.? Or is 7.0 imminent enough that we can wait for this? Adding this in will help us update to the latest Gradle version, as well as allow us to transition to plugins declarations.

eskatos commented on subprojects/plugin-use/src/integTest/groovy/org/gradle/plugin/repository/ResolvingWithPluginManagementSpec.groovy:
```diff
@@ -139,7 +139,35 @@ class ResolvingWithPluginManagementSpec extends AbstractDependencyResolutionTest
+        failure.assertHasDescription("Plugin [id: 'org.example.plugin', version: 'x'] was not found")
+    }
+
+    def 'when version range is specified, resolution sucseeds'() {
```
:x: Little typo
```suggestion
    def 'when version range is specified, resolution succeeds'() {
```

staale replied:
Fixed

eskatos commented on subprojects/plugin-use/src/integTest/groovy/org/gradle/plugin/repository/ResolvingWithPluginManagementSpec.groovy:
```diff
@@ -167,7 +195,35 @@ class ResolvingWithPluginManagementSpec extends AbstractDependencyResolutionTest
+        failure.assertHasDescription("Plugin [id: 'org.example.plugin', version: '1.2', artifact: 'org.example.plugin:plugin:x'] was not found")
+    }
+
+    def 'when artifact version range is specified, resolution sucseeds'() {
```
:x: Little typo
```suggestion
    def 'when artifact version range is specified, resolution succeeds'() {
```

staale replied:
Fixed

eskatos reviewed (changes requested):
Looking good. I requested two little changes for a typo.

eskatos commented:
@bot-gradle test this

bot-gradle commented:
<!-- {"replyTargetCommentId":775947848,"teamCityBuildId":"41200707","teamCityBuildHeadRef":"e11c958564f3f186baf18cb3da8d7138e868534f","teamCityBuildStatus":"PENDING","replyApproveReviewId":null} -->
OK, I've already triggered [ReadyForMerge build](https://builds.gradle.org/viewQueued.html?itemId=41200707) for you.

eskatos commented:
@Mergifyio update

bot-gradle commented:
<!-- {"replyTargetCommentId":null,"teamCityBuildId":"41367180","teamCityBuildHeadRef":"44b24f561b22e44900f87d93aa63be812f06355b","teamCityBuildStatus":"PENDING","replyApproveReviewId":null} -->
OK, I've already triggered [ReadyForNightly build](https://builds.gradle.org/viewQueued.html?itemId=41367180) for you.

eskatos commented:
@bot-gradle test this

bot-gradle commented:
<!-- {"replyTargetCommentId":776771797,"teamCityBuildId":"41368896","teamCityBuildHeadRef":"44b24f561b22e44900f87d93aa63be812f06355b","teamCityBuildStatus":"PENDING","replyApproveReviewId":null} -->
OK, I've already triggered [ReadyForMerge build](https://builds.gradle.org/viewQueued.html?itemId=41368896) for you.

jjohannes commented:
@bot-gradle test this

bot-gradle commented:
<!-- {"replyTargetCommentId":779652284,"teamCityBuildId":"41456593","teamCityBuildHeadRef":"ebb800d370bf23ec3f5059f8b6162af600d92d6a","teamCityBuildStatus":"PENDING","replyApproveReviewId":null} -->
OK, I've already triggered [ReadyForMerge build](https://builds.gradle.org/viewQueued.html?itemId=41456593) for you.
